src/Chess/ChessBoard.hs

```diff
@@ -30,8 +30,19 @@ chessBoardRowAsString rowNum chessBoard = "(" ++ (show rowNum) ++ ")  "  ++ (fol
 -- WARNING: Assumes given move is valid!
 -- TODO: implement + test this (1.5 hours) [Yiyi]
 makeMove :: ChessBoard -> ChessMove -> ChessBoard
-makeMove chessBoard move = chessBoard
+makeMove cb@(ChessBoard pieces state) (ChessMove from to)
+    | (isEmpty to cb) =
+        ChessBoard ((filter (\ (position, piece) -> piece /= pieceFrom) pieces) ++ [(to, pieceFrom)]) newState
```

Comment:
I think you'll have to compare `position` to `from` rather than `piece` and `pieceFrom`. Remember a Chessboard can have more than one of the same piece on the board (i.e. there can be two black bishops, 8 white pawns, etc.).

I recommend creating the following helper functions:

```haskell
-- Removes the piece at the given position from the ChessBoard if it exists
-- Does nothing if the position is empty
removePiece :: ChessPosition -> [(ChessPosition, ChessPiece)] -> [(ChessPosition, ChessPiece)]

-- Adds the given piece to the given position on the chessboard
-- Warning! Does not remove any pieces that may be at the same position
putPiece :: ChessPiece -> ChessPosition -> [(ChessPosition, ChessPiece)] -> [(ChessPosition, ChessPiece)]
```

Then writing `makeMove` becomes simple:

```haskell
makeMove cb@(ChessBoard pieces state) (ChessMove from to) = ChessBoard newPieces newState 
    where newPieces = putPiece pieceToMove to (removePiece from pieces)
               pieceToMove = getPieceAt from cb
               newState = if (isCheckMate cb (getPieceColour pieceFrom)) then Over else state
```
And if you need, feel free to edit the signature of `makeMove` so that the caller passes it the colour. Then you can simplify:

```haskell
makeMove cb@(ChessBoard pieces state) (ChessMove from to) colour = ChessBoard newPieces newState 
    where newPieces = putPiece pieceToMove to (removePiece from pieces)
               pieceToMove = getPieceAt from cb
               newState = if (isCheckMate cb colour) then Over else state
```